utils/S3UtilsMongoClient.js

```diff
@@ -207,132 +214,145 @@ class S3UtilsMongoClient extends MongoClientInterface {
-                                    collRes[metricLevel][resourceName] = {
-                                        ...baseMetricsObject,
-                                    };
-                                }
-                                collRes[metricLevel][resourceName][targetData] += data[metricLevel][resourceName];
-                                collRes[metricLevel][resourceName][targetCount]++;
-                                collRes[metricLevel][resourceName].deleteMarkerCount += res.value.isDeleteMarker ? 1 : 0;
-                            });
-                        }
-                    });
-                    Object.keys(data.account).forEach(account => {
-                        if (!collRes.account[account].locations) {
-                            collRes.account[account].locations = {};
-                        }
+                // Dynamically get the metrics based on the object state
+                if (isObjectCold) {
+                    targetCount += 'Cold';
+                    targetData += 'Cold';
+                } else if (isObjectRestoring) {
+                    targetCount += 'Restoring';
+                    targetData += 'Restoring';
+                } else if (isObjectRestored) {
+                    targetCount += 'Restored';
+                    targetData += 'Restored';
+                }
+
+                if (isMPUPart || isOverviewKey) {
+                    targetCount = 'mpuPartCount';
+                    targetData = 'mpuPartsData';
+                }
 
-                        Object.keys(data.location).forEach(location => {
-                            if (!collRes.account[account].locations[location]) {
-                                collRes.account[account].locations[location] = {
+                Object.keys(data).forEach(metricLevel => {
+                    // metricLevel can only be 'bucket', 'location' or 'account'
+                    if (validStorageMetricLevels.has(metricLevel)) {
+                        Object.keys(data[metricLevel]).forEach(resourceName => {
+                            // resourceName can be the name of bucket, location or account
+                            if (!collRes[metricLevel][resourceName]) {
+                                collRes[metricLevel][resourceName] = {
                                     ...baseMetricsObject,
                                 };
                             }
-                            collRes.account[account].locations[location][targetData] += data.location[location];
-                            collRes.account[account].locations[location][targetCount]++;
-                            collRes.account[account].locations[location].deleteMarkerCount += res.value.isDeleteMarker ? 1 : 0;
+                            collRes[metricLevel][resourceName][targetData] += data[metricLevel][resourceName];
+                            // Do not count the MPU parts as objects
```

Comment:
not an issue in this PR, but worth considering for the future: eventually, do we want to report MPU by number of uploads (i.e. "potential objects") or actual parts (which are stored internally as separate documents, and actually each reference some data)...

thinking about this,
* it may actually be better to count each part as an object : so we can also report on left-over parts even if the overview key is missing. Semantics may not be so good (part vs object), but I'd rather report something that more closely matches the storage than the user's business logic for now, and thus not mask any issue.
* as far as semantics go, `mpuPartsCount` should be a number of parts: if we count only the overview keys, should be something like `mpuUploadsCount` instead

Reply:
> not an issue in this PR, but worth considering for the future

I formally disagree that more effort should be put into this obsolete script. But it's worth discussing how to count MPUs for Scuba.